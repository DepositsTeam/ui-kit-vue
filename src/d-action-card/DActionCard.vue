<template>
  <d-box class="d-action-card">
    <d-box class="d-action-card__header">
      <d-box is="img" class="d-action-card__logo" :src="logo" :alt="title" />
      <d-text class="d-action-card__header__title" font-face="heroNew">{{
        title
      }}</d-text>
      <d-text
        class="d-action-card__header__description"
        font-face="circularSTD"
        >{{ description }}</d-text
      >
    </d-box>
    <d-box class="d-action-card__body">
      <slot name="buttons">
        <d-auto-layout>
          <d-button
            size="huge"
            colorScheme="primary"
            @click="emit('button1-clicked')"
            :disabled="disable"
          >
            {{ button1Text }}
          </d-button>
          <d-button
            size="huge"
            :right-icon="ExternalLinkIcon"
            colorScheme="neutral"
            @click="emit('button2-clicked')"
          >
            {{ button2Text }}
          </d-button>
        </d-auto-layout>
      </slot>
    </d-box>
  </d-box>
</template>

<script setup>
import { DBox, DText, DButton, ExternalLinkIcon } from "@/main";
import DAutoLayout from "@/d-auto-layout/DAutoLayout.vue";

const emit = defineEmits(["button1-clicked", "button2-clicked"]);
defineProps({
  title: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  logo: {
    type: String,
    required: true,
  },
  hasLearnMore: {
    type: Boolean,
    default: true,
  },
  disable: {
    type: Boolean,
    default: false,
  },
  button1Text: {
    type: String,
    default: "Connect",
  },
  button2Text: {
    type: String,
    default: "Learn More",
  },
});
</script>

<style scoped lang="scss">
.d-action-card {
  display: flex;
  padding: 16px;
  flex-direction: column;
  justify-content: center;
  gap: 24px;
  border-radius: 8px;
  border: 0.5px solid #e1e7ec;

  .d-action-card__header {
    display: flex;
    flex-direction: column;
    gap: 8px;

    .d-action-card__logo {
      width: 32px;
      height: 32px;
      border-radius: 50%;
    }

    .d-action-card__header__title {
      color: #1b1e21;
      font-size: 16px;
      font-style: normal;
      font-weight: 500;
      line-height: normal;
      letter-spacing: 0.16px;
    }

    .d-action-card__header__description {
      color: #6d7786;
      font-size: 14px;
      font-style: normal;
      font-weight: 450;
      line-height: normal;
      letter-spacing: 0.14px;
    }
  }

  .d-action-card__body {
    display: flex;
    padding-top: 16px;
    align-items: flex-start;
    gap: 8px;
    align-self: stretch;
    border-top: 0.5px solid #e1e7ec;
  }
}
</style>
