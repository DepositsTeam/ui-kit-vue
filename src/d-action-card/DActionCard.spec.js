import { mount } from "@vue/test-utils";
import DActionCard from "./DActionCard.vue";
import { expect, test } from "vitest";

test("renders the component", async () => {
  expect(DActionCard).to.not.be.undefined;

  const wrapper = mount(DActionCard, {
    props: {
      title: "Quickbooks",
      description:
        "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically.",
      logo: "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn",
      button1Text: "",
      button2Text: "",
    },
  });

  expect(wrapper.get(".d-action-card__header__title").text()).to.be.equal(
    "Quickbooks"
  );

  expect(wrapper.get(".d-action-card__header__description").text()).to.be.equal(
    "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically."
  );

  expect(wrapper.get(".d-action-card__logo").attributes("src")).to.be.equal(
    "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn"
  );

  expect(wrapper.get(".d-action-card__logo").attributes("alt")).to.be.equal(
    "Quickbooks"
  );

  const buttons = wrapper.findAllComponents({ name: "DButton" });

  expect(buttons).to.have.lengthOf(0);
});

test("matches snapshot", async () => {
  const wrapper = mount(DActionCard);
  expect(wrapper.element).toMatchSnapshot();
});

test("renders the component with buttons", async () => {
  const wrapper = mount(DActionCard, {
    props: {
      title: "Quickbooks",
      description:
        "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically.",
      logo: "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn",
      button1Text: "Connect",
      button2Text: "Learn more",
    },
  });

  const buttons = wrapper.findAllComponents({ name: "DButton" });

  expect(buttons).to.have.lengthOf(2);
});

test("triggers events on click", async () => {
  const wrapper = mount(DActionCard, {
    props: {
      title: "Quickbooks",
      description:
        "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically.",
      logo: "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn",
      button1Text: "Connect",
      button2Text: "Learn more",
    },
  });

  const buttons = wrapper.findAllComponents({ name: "DButton" });

  await buttons[0].trigger("click");

  let emitted = wrapper.emitted("button1Clicked");

  expect(emitted).to.have.lengthOf(1);

  await buttons[1].trigger("click");

  emitted = wrapper.emitted("button2Clicked");

  expect(emitted).to.have.lengthOf(1);
});

test("it renders slots appropriately", async () => {
  const wrapper = mount(DActionCard, {
    props: {
      title: "Quickbooks",
      description:
        "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically.",
      logo: "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn",
      button1Text: "Connect",
      button2Text: "Learn more",
    },
    slots: {
      title: "<div id='slottedTitle'>Slotted title</div>",
      description: "<div id='slottedDescription'>Slotted description</div>",
      buttons: "<button>New button</button>",
    },
  });

  expect(wrapper.find(".d-action-card__header__title").exists()).to.be.false;

  expect(wrapper.find(".d-action-card__header__description").exists()).to.be
    .false;

  expect(wrapper.get("#slottedTitle").text()).to.be.equal("Slotted title");

  expect(wrapper.get("#slottedDescription").text()).to.be.equal(
    "Slotted description"
  );

  const buttons = wrapper.findAllComponents({ name: "DButton" });

  expect(buttons).to.have.lengthOf(0);

  expect(wrapper.get("button").text()).to.be.equal("New button");
});

test("individual button slots are rendered appropriately", async () => {
  const wrapper = mount(DActionCard, {
    props: {
      title: "Quickbooks",
      description:
        "Connect Renapay to your QuickBooks account to allow seamles synchronisation of your invoices, payments and customers to QuickBooks automatically.",
      logo: "https://play-lh.googleusercontent.com/ahJtMe0vfOlAu1XJVQ6rcaGrQBgtrEZQefHy7SXB7jpijKhu1Kkox90XDuH8RmcBOXNn",
      button1Text: "Connect",
      button2Text: "Learn more",
    },
    slots: {
      button1: "<button id='slottedButton1'>New button</button>",
      button2: "<button id='slottedButton2'>New button</button>",
    },
  });

  const buttons = wrapper.findAllComponents({ name: "DButton" });

  expect(buttons).to.have.lengthOf(0);

  expect(wrapper.find("#slottedButton1").exists()).to.be.true;
  expect(wrapper.find("#slottedButton2").exists()).to.be.true;
});
