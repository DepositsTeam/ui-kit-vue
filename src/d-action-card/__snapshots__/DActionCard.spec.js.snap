// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="circularSTD deposits-ui-box d-action-card"
  data-ui-kit-auto_generated-cltvdguj3s=""
  data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
  data-v-441f4658=""
>
  
  <div
    class="circularSTD deposits-ui-box d-action-card__header"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-441f4658=""
  >
    
    
    <img
      class="circularSTD deposits-ui-box d-action-card__logo"
      data-ui-kit-auto_generated-crmlexksxk=""
      data-ui-kit-theming-rules-auto_generated-crmlexksxk=""
      data-v-441f4658=""
    >
      
      
    </img>
    
    
    <p
      center="false"
      class="heroNew deposits-ui-box ui-text my0 d-action-card__header__title d-action-card__header__title"
      data-ui-kit-auto_generated-m7u0cw2pdt=""
      data-ui-kit-theming-rules-auto_generated-m7u0cw2pdt=""
      data-v-441f4658=""
      equallineheight="false"
      mb0="false"
      ml0="false"
      mr0="false"
      mt0="false"
      mx0="false"
      my0="true"
      uppercase="false"
    >
      
      
      
      
      
    </p>
    
    
    <p
      center="false"
      class="circularSTD deposits-ui-box ui-text my0 d-action-card__header__description d-action-card__header__description"
      data-ui-kit-auto_generated-rkysuan9l1=""
      data-ui-kit-theming-rules-auto_generated-rkysuan9l1=""
      data-v-441f4658=""
      equallineheight="false"
      mb0="false"
      ml0="false"
      mr0="false"
      mt0="false"
      mx0="false"
      my0="true"
      uppercase="false"
    >
      
      
      
      
      
    </p>
    
    
  </div>
  <div
    class="circularSTD deposits-ui-box d-action-card__body"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-441f4658=""
  >
    
    
    <div
      class="circularSTD deposits-ui-box d-auto-layout direction__horizontal"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-441f4658=""
      style="--spacing-value: 10px;"
    >
      
      
      
      <button
        class="heroNew deposits-ui-box ui-button semantic__primary size__large"
        data-test="button1"
        data-ui-kit-auto_generated-c1xbggyioe=""
        data-ui-kit-theming-rules-auto_generated-c1xbggyioe=""
        data-v-441f4658=""
        data-v-5100a3c9=""
      >
        
        <div
          class="circularSTD deposits-ui-box"
          data-ui-kit-auto_generated-wh6g3feqwh=""
          data-ui-kit-theming-rules-auto_generated-wh6g3feqwh=""
          data-v-5100a3c9=""
        >
          
          <!--v-if-->
          <div
            class="circularSTD deposits-ui-box ui-button__button-text"
            data-ui-kit-auto_generated-cltvdguj3s=""
            data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
            data-v-5100a3c9=""
          >
            
            <div
              class="circularSTD deposits-ui-box"
              data-ui-kit-auto_generated-cltvdguj3s=""
              data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
              data-v-5100a3c9=""
            >
              
              
              Connect
              
              
            </div>
            
          </div>
          <!--v-if-->
          <!--v-if-->
          
        </div>
        
      </button>
      
      
      <button
        class="heroNew deposits-ui-box ui-button semantic__default size__large"
        data-test="button2"
        data-ui-kit-auto_generated-c1xbggyioe=""
        data-ui-kit-theming-rules-auto_generated-c1xbggyioe=""
        data-v-441f4658=""
        data-v-5100a3c9=""
      >
        
        <div
          class="circularSTD deposits-ui-box"
          data-ui-kit-auto_generated-wh6g3feqwh=""
          data-ui-kit-theming-rules-auto_generated-wh6g3feqwh=""
          data-v-5100a3c9=""
        >
          
          <!--v-if-->
          <div
            class="circularSTD deposits-ui-box ui-button__button-text"
            data-ui-kit-auto_generated-cltvdguj3s=""
            data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
            data-v-5100a3c9=""
          >
            
            <div
              class="circularSTD deposits-ui-box"
              data-ui-kit-auto_generated-cltvdguj3s=""
              data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
              data-v-5100a3c9=""
            >
              
              
              Learn More
              
              
            </div>
            
          </div>
          <!--v-if-->
          <svg
            class="deposits-ui-box ui-button__right-icon"
            data-ui-kit-auto_generated-7LxGHMe8z3=""
            data-ui-kit-theming-rules-auto_generated-7LxGHMe8z3=""
            data-v-5100a3c9=""
            fill="none"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M14.1676 4.00171H20V10.4174M19.9964 14.7V18.6662C19.9964 19.0198 19.856 19.3587 19.606 19.6087C19.356 19.8587 19.0169 19.9994 18.6633 19.9994H5.33315C4.97958 19.9994 4.64044 19.8587 4.39043 19.6087C4.14042 19.3587 4 19.0198 4 18.6662V5.33586C4 4.98229 4.14042 4.64337 4.39043 4.39336C4.64044 4.14335 4.97958 4.00293 5.33315 4.00293H10.1657M11.9987 11.9987L19.9973 4"
              stroke="currentcolor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            
          </svg>
          
        </div>
        
      </button>
      
      
      
    </div>
    
    
  </div>
  
</div>
`;
